lib/elixir/lib/exception.ex

```diff
@@ -511,35 +520,49 @@ defmodule SystemLimitError do
 end
 
 defmodule SyntaxError do
-  defexception [file: nil, line: nil, description: "syntax error"]
+  defexception [file: nil, line: nil, description: "syntax error", message: nil]
```

Comment:
you are right, it is suitable for a situation like Code.LoadError https://github.com/elixir-lang/elixir/pull/3647/files#diff-548f41b79339d31106664af73b3c8a19R678